### PR TITLE
Minor `PostMergeBlocksSyncPeerAllocationStrategy` refactor

### PR DESCRIPTION
**Pull request by asdacap**
Extract speed related logic to BySpeedStrategy.

## Changes:
- Extract speed related logic to BySpeedStrategy.
- Added some unit test.

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [X] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Other (please describe): 

## Testing
**Requires testing**

- [X] Yes
- [ ] No

**In case you checked yes, did you write tests??**

- [X] Yes
- [ ] No

**Comments about testing , should you have some** (optional)

Goerli can sync to head

## Further comments (optional)

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by dceleda** [comment]
Probably needs to be rabased (157 modifications)

**Comment by asdacap** [comment]
Rebased, should be only 3 files.

**Review by MarekM25 — COMMENTED** [review]
Looks ok. We can merge it to develop for sure :) but let's make sure we run smoke tests before merging to master

File: src/Nethermind/Nethermind.Merge.Plugin/Synchronization/PostMergeBlocksSyncPeerAllocationStrategy.cs [hunk 1]
```diff
@@ -40,66 +43,27 @@ public PostMergeBlocksSyncPeerAllocationStrategy(long? minBlocksAhead, IBeaconPi
...
                 {
                     // we need to guarantee the peer can have all the block prior to beacon pivot
-                    continue;
+                    return false;
                 }
-            } else if (info.HeadNumber < (blockTree.BestSuggestedBody?.Number ?? 0) + (_minBlocksAhead ?? 1)) {
-               continue;
             }
-
-            long averageTransferSpeed = GetSpeed(nodeStatsManager, info) ?? 0;
```
Comment (MarekM25) [inline src/Nethermind/Nethermind.Merge.Plugin/Synchronization/PostMergeBlocksSyncPeerAllocationStrategy.cs:84]: hmmm did we remove it?

**Comment by asdacap** [comment]
I dont thing anything will go to master anymore

**Review by MarekM25 — APPROVED** [review]

File: src/Nethermind/Nethermind.Merge.Plugin/Synchronization/PostMergeBlocksSyncPeerAllocationStrategy.cs (hunk 1 above)
Comment (asdacap) [reply src/Nethermind/Nethermind.Merge.Plugin/Synchronization/PostMergeBlocksSyncPeerAllocationStrategy.cs:84]: It all goes to BySpeedStrategy

**Pull request merged**